File: python/agentic_commerce_protocol/models/error_type.py
# this file is @generated
from enum import Enum


class ErrorType(str, Enum):
    INVALID_REQUEST = "invalid_request"
    REQUEST_NOT_IDEMPOTENT = "request_not_idempotent"
    PROCESSING_ERROR = "processing_error"
    SERVICE_UNAVAILABLE = "service_unavailable"

    def __str__(self) -> str:
        return str(self.value)

File: ruby/lib/agentic_commerce_protocol/models/error_type.rb
# frozen_string_literal: true
# This file is @generated
module AgenticCommerceProtocol
  class ErrorType
    INVALID_REQUEST = "invalid_request".freeze
    REQUEST_NOT_IDEMPOTENT = "request_not_idempotent".freeze
    PROCESSING_ERROR = "processing_error".freeze
    SERVICE_UNAVAILABLE = "service_unavailable".freeze

    def self.all_vars
      @all_vars ||= [INVALID_REQUEST, REQUEST_NOT_IDEMPOTENT, PROCESSING_ERROR, SERVICE_UNAVAILABLE].freeze
    end

    def initialize(value)
      unless ErrorType.all_vars.include?(value)
        raise "Invalid ENUM value '#{value}' for class #ErrorType"
      end

      @value = value
    end

    def self.deserialize(value)
      return value if ErrorType.all_vars.include?(value)
      raise "Invalid ENUM value '#{value}' for class #ErrorType"
    end

    def serialize
      @value
    end
  end
end
